core/internal/src/mill/internal/LinePrefixOutputStream.scala

```diff
@@ -42,7 +42,10 @@ private[mill] class LinePrefixOutputStream(
       val bufferString = buffer.toString
       if (bufferString.length > 0) {
         val s = fansi.Str.apply(bufferString, errorMode = fansi.ErrorMode.Sanitize)
-        endOfLastLineColor = s.getColor(math.max(0, s.length - 1))
+        if (s.length > 0) {
+          // only change last value if we have some proper ansi string
+          endOfLastLineColor = s.getColor(s.length - 1)
```

Comment:
If the string a length of zero, there is also no color information as index `0`. 

Now, we only attempt to get the color of the last string position, if there is a character at all. Otherwise we don't update the color info.